Added pre-commit hooks (#498)

COUR4G3 opened the pull request:
Added pre-commit hooks for checking and updating news in projects using pre-commit (#498)

* added ``towncrier-check`` to just check newly commited news fragments
* added ``towncrier-update`` to actually update the news
* added documentation section for ``pre-commit``

# Description

Added a pre-commit hooks for checking and updating news in projects using pre-commit, as suggested in my issue #498.

# Checklist
<!-- add a "X" inside the brackets to confirm -->
* [ ] Make sure changes are covered by existing or new tests.
* [ ] For at least one Python version, make sure local test run is green.
* [X] Create a file in `src/towncrier/newsfragments/`. Describe your
  change and include important information. Your change will be included in the public release notes.
* [ ] Make sure all GitHub Actions checks are green (they are automatically checking all of the above).
* [ ] Ensure `docs/tutorial.rst` is still up-to-date.
* [ ] If you add new **CLI arguments** (or change the meaning of existing ones), make sure `docs/cli.rst` reflects those changes.
* [ ] If you add new **configuration options** (or change the meaning of existing ones), make sure `docs/configuration.rst` reflects those changes.

COUR4G3 committed:
Added pre-commit hooks for checking and updating news in projects using pre-commit (#498)

* added ``towncrier-check`` to just check newly commited news fragments
* added ``towncrier-update`` to actually update the news
* added documentation section for ``pre-commit``

asottile commented on docs/pre-commit.rst:
```diff
@@ -0,0 +1,40 @@
+
+    repos:
+      - repo: https://github.com/twisted/towncrier
+      - rev: trunk
```
I would not recommend this -- as [per the docs](https://pre-commit.com/#using-the-latest-version-for-a-repository)

I either put the latest tag here or some placeholder invalid value with a comment indicating the user should run `pre-commit autoupdate` after adding

hynek replied:
Thanks Anthony! Would you mind fixing this @COUR4G3?

COUR4G3 replied:
Well spotted @asottile 😄 

Sure, will fix this up!

COUR4G3 replied:
I replaced `trunk` with the **next** minor release `22.13.0`, assuming this is included in the next minor release. And I added a note about `pre-commit autoupdate`.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

COUR4G3 commented:
I replaced `trunk` with the **next** minor release `22.13.0`, assuming this is included in the next minor release. And I added a note about `pre-commit autoupdate`.

COUR4G3 committed:
Merge branch 'trunk' into 498-pre-commit-hooks

hynek commented on docs/pre-commit.rst:
```diff
@@ -0,0 +1,40 @@
+pre-commit
+==========
+
+``towncrier`` can also be used in your `pre-commit <https://pre-commit.com/>`_ configuration (``.pre-commit-config.yaml``)
+to check and/or update your news fragments on commit or during CI processes.
```
```suggestion
``towncrier`` can also be used in your `pre-commit <https://pre-commit.com/>`_ configuration (``.pre-commit-config.yaml``) to check and/or update your news fragments on commit or during CI processes.
```

hynek commented on docs/pre-commit.rst:
```diff
@@ -0,0 +1,40 @@
+``towncrier`` can also be used in your `pre-commit <https://pre-commit.com/>`_ configuration (``.pre-commit-config.yaml``)
+to check and/or update your news fragments on commit or during CI processes.
+
+No additional configuration is needed in your ``towncrier`` configuration, the hook will read from the appropriate configuration
+files in your project.
```
```suggestion
No additional configuration is needed in your ``towncrier`` configuration; the hook will read from the appropriate configuration files in your project.
```

(I believe this was comma splice?)

hynek commented on docs/pre-commit.rst:
```diff
@@ -0,0 +1,40 @@
+Usage with custom configuration and directories
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+News fragments are stored in ``changelog.d/`` in the root of the repository and we want to keep the news fragments when
+running ``update``:
```
```suggestion
News fragments are stored in ``changelog.d/`` in the root of the repository and we want to keep the news fragments when running ``update``:
```

hynek reviewed (changes requested):
Thanks! We use semantic linefeeds for docs but until #500 is merged it's not documented so I did it inline.

hynek committed:
Apply suggestions from code review

hynek committed:
Merge branch 'trunk' into 498-pre-commit-hooks

hynek committed:
Merge branch 'trunk' into 498-pre-commit-hooks
